ServiceLayer/Services.cs

```diff
@@ -0,0 +1,107 @@
+﻿using DAL;
+using DAL.Model;
+using System;
+using System.Linq;
+
+namespace ServiceLayer
+{
+    public class Services
+    {
+        private readonly BookRepository _BookRepository;
+
+        public Services(BookRepository bookRepository)
+        {
+            _BookRepository = bookRepository;
+        }
+
+
+        public Result GetBooks()
+        {
+            var books = _BookRepository.GetAllBooks();
+            Result result = new Result();
+            if (books.Count == 0)
+            {
+
+                result.ErrorMessage = "No Books Found";
+                result.StatusCode = 404;
+                result.Value = null;
+                
+            }
+            else
+            {
+                result.StatusCode = 200;
+                result.Value = books;
+            }
+            return result;
+        }
+        public Result GetBookNameById(int id)
+        {
+            Result result = new Result();
+            if(id<0)
+            {
+                result.ErrorMessage = "Invalid Id, Id should be a positive number";
+                result.StatusCode = 400;
+                return result;
+            }
+            var book = _BookRepository.GetById(id);
+            if (book == null)
+            {
+                result.ErrorMessage = "ID Not Found";
+                result.StatusCode = 404;
+                return result;
+            }
+            else
+            {
+                result.Value = book;
+                return result;
+            }
+                
+
+
+        }
+
+        public bool Update(Book book)
+        {
+            return _BookRepository.ReplaceEmployee(book);
```

Comment:
check first the book that needs to be updated is present in the list or not. If not present return error

Reply:
I did the same ....You reviewed my first commit